internal/pipeline/parser_test.go

```diff
@@ -106,23 +106,110 @@ steps:
 			&CommandStep{
 				Command: "docker build .",
 				RemainingFields: map[string]any{
-					"agents": map[string]any{
-						"queue": "default",
-					},
+					"agents": ordered.MapFromItems(
+						ordered.TupleSA{Key: "queue", Value: "default"},
+					),
 					"name":              ":docker: building image",
 					"type":              "script",
 					"agent_query_rules": []any{"queue=default"},
 				},
 			},
 		},
 		RemainingFields: map[string]any{
-			"base_step": map[string]any{
-				"type":              "script",
-				"agent_query_rules": []any{"queue=default"},
+			"base_step": ordered.MapFromItems(
+				ordered.TupleSA{Key: "type", Value: "script"},
+				ordered.TupleSA{Key: "agent_query_rules", Value: []any{"queue=default"}},
+			),
+		},
+	}
+	if diff := cmp.Diff(got, want, cmp.Comparer(ordered.EqualSA)); diff != "" {
+		t.Errorf("parsed pipeline diff (-got +want):\n%s", diff)
+	}
+
+	gotJSON, err := json.MarshalIndent(got, "", "  ")
+	if err != nil {
+		t.Fatalf(`json.MarshalIndent(got, "", "  ") error = %v`, err)
+	}
+
+	const wantJSON = `{
+  "base_step": {
+    "type": "script",
+    "agent_query_rules": [
+      "queue=default"
+    ]
+  },
+  "steps": [
+    {
+      "agent_query_rules": [
+        "queue=default"
+      ],
+      "agents": {
+        "queue": "default"
+      },
+      "command": "docker build .",
+      "name": ":docker: building image",
+      "type": "script"
+    }
+  ]
+}`
+	if diff := cmp.Diff(string(gotJSON), wantJSON); diff != "" {
+		t.Errorf("marshalled JSON diff (-got +want):\n%s", diff)
+	}
+}
+
+func TestParserSupportsDoubleMerge(t *testing.T) {
+	const complexYAML = `---
+base_step: &base_step
+  type: script
+  agent_query_rules:
+    - queue=default
+
+remainder: &remainder
+  name: ':docker: building image'
+  command: docker build .
+  agents:
+    queue: default
+
+steps:
+  - <<: *base_step
+    <<: *remainder
+`
```

Comment:
Can we get a test for the official way to do it as well?
```yaml
steps:
- <<: [*base_step, *remainder]
```

Reply:
Done!